iree/samples/simple_embedding/simple_embedding_run.c

```diff
@@ -0,0 +1,229 @@
+  IREE_RETURN_IF_ERROR(iree_vm_context_create_with_modules(
+      instance, &modules[0], kVmModuleNum, iree_allocator_system(), &context));
+  iree_vm_module_release(hal_module);
+  iree_vm_module_release(bytecode_module);
+
+  // Lookup the entry point function.
+  // Note that we use the synchronous variant which operates on pure type/shape
+  // erased buffers.
+  const char kMainFunctionName[] = "module.simple_mul";
+  iree_vm_function_t main_function;
+  IREE_RETURN_IF_ERROR(iree_vm_context_resolve_function(
+      context, iree_make_cstring_view(kMainFunctionName), &main_function));
+
+  // Allocate buffers that can be mapped on the CPU and that can also be used
+  // on the device. Not all devices support this, but the ones we have now do.
+  const int kElementCount = 4;
+  iree_hal_buffer_t* arg0_buffer = NULL;
+  iree_hal_buffer_t* arg1_buffer = NULL;
+  iree_hal_memory_type_t input_memory_type =
+      IREE_HAL_MEMORY_TYPE_HOST_LOCAL | IREE_HAL_MEMORY_TYPE_DEVICE_VISIBLE;
+  IREE_RETURN_IF_ERROR(iree_hal_allocator_allocate_buffer(
+      iree_hal_device_allocator(device), input_memory_type,
+      IREE_HAL_BUFFER_USAGE_ALL, sizeof(float) * kElementCount, &arg0_buffer));
+  IREE_RETURN_IF_ERROR(iree_hal_allocator_allocate_buffer(
+      iree_hal_device_allocator(device), input_memory_type,
+      IREE_HAL_BUFFER_USAGE_ALL, sizeof(float) * kElementCount, &arg1_buffer));
+
+  // Populate initial values for 4 * 2 = 8.
+  const float kFloat4 = 4.0f;
+  const float kFloat2 = 2.0f;
+  IREE_RETURN_IF_ERROR(iree_hal_buffer_fill(arg0_buffer, 0, IREE_WHOLE_BUFFER,
+                                            &kFloat4, sizeof(float)));
+  IREE_RETURN_IF_ERROR(iree_hal_buffer_fill(arg1_buffer, 0, IREE_WHOLE_BUFFER,
+                                            &kFloat2, sizeof(float)));
+
+  // Wrap buffers in shaped buffer views.
+  iree_hal_dim_t shape[1] = {kElementCount};
+  iree_hal_buffer_view_t* arg0_buffer_view = NULL;
+  iree_hal_buffer_view_t* arg1_buffer_view = NULL;
+  IREE_RETURN_IF_ERROR(iree_hal_buffer_view_create(
+      arg0_buffer, IREE_HAL_ELEMENT_TYPE_FLOAT_32, shape, IREE_ARRAYSIZE(shape),
+      &arg0_buffer_view));
+  IREE_RETURN_IF_ERROR(iree_hal_buffer_view_create(
+      arg1_buffer, IREE_HAL_ELEMENT_TYPE_FLOAT_32, shape, IREE_ARRAYSIZE(shape),
+      &arg1_buffer_view));
+  iree_hal_buffer_release(arg0_buffer);
+  iree_hal_buffer_release(arg1_buffer);
+
+  // Setup call inputs with our buffers. The vm list is allocated statically.
+  iree_vm_list_t* inputs = NULL;
+  iree_host_size_t input_list_size = iree_vm_list_storage_size(
+      /*element_type=*/NULL, /*capacity=*/2);
+  void* input_list_storage = iree_alloca(input_list_size);
+  IREE_RETURN_IF_ERROR(
+      iree_vm_list_initialize(
+          iree_make_byte_span(input_list_storage, input_list_size),
+          /*element_type=*/NULL,
+          /*capacity=*/2, &inputs),
+      "can't allocate input vm list");
```

Comment:
Please comment on the iree_vm_list_t allocation. Other examples use `iree_vm_list_create` instead; however, `iree_vm_list_initialize` handles statically allocated list.

Reply:
(@benvanik , I'm not sure either)

Reply:
iree_vm_list_create is the easiest/safest way to do things and is much less code, so I'd say switch to using that - this is not the hot path and I feel like the simplicity in the example is a good idea

Reply:
Done. Thanks for the suggestion!